x/programs/runtime/storage.go

```diff
@@ -0,0 +1,121 @@
+package runtime
+
+import (
+	"encoding/binary"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/hypersdk/consts"
+)
+
+const (
+	HRP                = "simulator_key_"
+	programPrefix      = 0x0
+	programCountPrefix = 0x1
+	storagePrefix      = 0x2
+)
+
+func ProgramKey(asset uint64) (k []byte) {
+	k = make([]byte, 1+consts.IDLen)
+	// convert uint64 to bytes
+	binary.BigEndian.PutUint64(k[1:], asset)
+	k[0] = programPrefix
+	return
+}
+
+// StorageKey returns the key used to store a value in the program's storage
+// [storagePrefix] + [programID] + [key]
+func StorageKey(programID uint64, key []byte) (k []byte) {
+	k = make([]byte, 1+consts.Int64Len+len(key))
+	k[0] = storagePrefix
+	binary.BigEndian.PutUint64(k[1:], programID)
+	copy(k[1+consts.Int64Len:], key)
+	return k
+}
+
+// ProgramCountKey returns the key used to store the program count
+func ProgramCountKey() []byte {
+	return []byte{programCountPrefix}
+}
+
+// IncrementProgramCount increments the program count by 1
+func IncrementProgramCount(db database.Database) error {
+	countKey := ProgramCountKey()
+	bytes, err := db.Get(countKey)
+	var count uint64
+	if err != nil {
```

Comment:
this should be checking for not found else return error or just use `GetProgramCount`